### DISCUSSION TITLE
How do I change colors variables dynamically

### DISCUSSION THREAD

**Discussion by itancc** [unocss/unocss#1083] [Q&A] [answered]
：When I enter a color in the input box, how does **text-primary** and **bg-primary** automatically change to the input color

![image](https://user-images.githubusercontent.com/46319251/172848304-93669ff4-78bb-45dc-9641-7abd168381b7.png)

Online instance： [Online instance](https://stackblitz.com/edit/vitejs-vite-5uradm?file=src/components/HelloWorld.vue)

thank you!

**Comment by zyyv** [comment, marked as answer]
i guess you want to use dynamic class, but now it's not surpport

**Reply by itancc** [reply]
Thank you for your help. In short, is there a way to dynamically change the variables in the theme? I read about @unocss/ Runtime, but I'm not sure how this package works
![image](https://user-images.githubusercontent.com/46319251/172970036-6283ac84-5651-4ace-a21e-d4bb73569f67.png)

**Reply by chu121su12** [reply]
As far as I know, using unocss in compilation mode only produces 1 set of css for the application. When deployed, usually you no longer have the engine with you.

Runtime is the opposite, which is bringing then whole unocss code in the runtime of your application. There's overhead since the runtime must be downloaded and executed on every page load but since it is available at the *runtime*, you can modify the config and have the injected css updated.

**Reply by itancc** [reply]
Ok, I see. I will read the source code，Thank you！
